app/controllers/shelters_controller.rb

```diff
@@ -22,7 +26,17 @@ def pets
   end
 
   def show
-    @shelter = Shelter.find(params[:id])
+    if params[:admin] == "admin"
+      @admin = true
+      @current_shelter = Shelter.find(params[:id])
+      @shelter = Shelter.name_and_address(params[:id])
+      @pending_pets = Shelter.find(params[:id]).pending_pets
+      @age_stats = Shelter.age_stats(@current_shelter)
+      @pet_count = Shelter.pet_count(@current_shelter)
+      @pets_with_homes = Shelter.pets_with_homes(@current_shelter)
+    else
+      @shelter = Shelter.find(params[:id])
+    end
```

Comment:
There is way too much going on in here. We realize that the extensions are doing a lot but this is a `show` action so we are just looking for a single instance of a shelter. So line 31 is finding the shelter, now in your view you can call all of these methods on that one instance vs creating all of these additional instance variables in the controller. 
Also these methods (`name_and_address`, `pending_pets`, `age_stats`, `pet_count`, etc) should not be class methods but instance methods. So you would need to change that in your `shelter` model.